### PR TITLE
Only enable knative trait when there is a knative endpoint

### PR DESCRIPTION
**Pull request by claudio4j**
https://github.com/apache/camel-k/issues/5145

Remove the SortChannelFakeClient function since it's not used.

**Release Note**
```release-note
Only enable knative trait when there is a knative endpoint
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: pkg/trait/knative.go [hunk 1]
```diff
@@ -70,7 +71,8 @@ func (t *knativeTrait) Configure(e *Environment) (bool, *TraitCondition, error)
 	if e.Integration == nil {
 		return false, nil, nil
 	}
-	if !pointer.BoolDeref(t.Enabled, true) {
```
Comment (squakez) [inline pkg/trait/knative.go:73]: Why do you need this?

File: pkg/trait/knative.go [hunk 2]
```diff
@@ -81,6 +83,24 @@ func (t *knativeTrait) Configure(e *Environment) (bool, *TraitCondition, error)
 		if err != nil {
 			return false, nil, err
 		}
+
+		if t.Enabled == nil {
```
Comment (squakez) [inline pkg/trait/knative.go:87]: I think this is what the different `filterMetaItems` calls after this piece of code do. At the end of the `configure` you should have all the knative endpoints in the ChannelSources, ChannelSinks, ... variable. I think it is not convenient calculating this twice. Maybe, you can check those resources after they have been collected and do the needed logic accordingly.

File: pkg/trait/knative.go (hunk 1 above)
Comment (claudio4j) [reply pkg/trait/knative.go:73]: I noticed the `NewIntegrationConditionUserDisabled`  says the trait is explicitly disabled, so this sets the condition only when the user sets `enabled=false`.

File: pkg/trait/knative.go (hunk 2 above)
Comment (claudio4j) [reply pkg/trait/knative.go]: I noticed it, but wanted to have a check before, as these filters are intended to check for the specifics of knative. Anyway, I changed the check to account for the results of the knative endpoints.

File: pkg/trait/knative_test.go [hunk 3]
```diff
@@ -348,6 +348,155 @@ func TestKnativePlatformHttpDependencies(t *testing.T) {
...
+	require.NoError(t, knTrait.Apply(&environment))
+
+	// call the post step processors
+	for _, processor := range environment.PostStepProcessors {
+		assert.Nil(t, processor(&environment))
+	}
+
+	assert.True(t, *knTrait.Enabled)
+	assert.Contains(t, environment.Integration.Status.Capabilities, v1.CapabilityKnative)
+	assert.Contains(t, environment.Integration.Status.Dependencies, "mvn:org.apache.camel.k:camel-k-knative-impl")
```
Comment (squakez) [inline pkg/trait/knative_test.go:424]: I'd remove this as it not necessarily true. This dependency has to be driven by the catalog and may be subjected to chages. The check on the presence of Capability should be enough.

File: pkg/trait/knative_test.go [hunk 4]
```diff
@@ -348,6 +348,155 @@ func TestKnativePlatformHttpDependencies(t *testing.T) {
...
+	assert.False(t, ok)
+	assert.Nil(t, condition)
+
+	// call the post step processors
+	for _, processor := range environment.PostStepProcessors {
+		assert.Nil(t, processor(&environment))
+	}
+
+	assert.NotContains(t, environment.Integration.Status.Capabilities, v1.CapabilityKnative)
+	assert.NotContains(t, environment.Integration.Status.Dependencies, "mvn:org.apache.camel.k:camel-k-knative-impl")
```
Comment (squakez) [inline pkg/trait/knative_test.go:496]: Ditto

File: pkg/trait/knative.go (hunk 1 above)
Comment (squakez) [reply pkg/trait/knative.go:73]: But I think that's exactly what `!pointer.BoolDeref(t.Enabled, true)` does. If the uses does not set explicitly to `false`, then it is always  `true`.

File: pkg/trait/knative_test.go [hunk 5]
```diff
@@ -348,6 +348,155 @@ func TestKnativePlatformHttpDependencies(t *testing.T) {
...
+	require.NoError(t, init.Apply(&environment))
+
+	// configure the knative trait
+	knTrait, _ := newKnativeTrait().(*knativeTrait)
+	ok, condition, err = knTrait.Configure(&environment)
+	require.NoError(t, err)
+	assert.False(t, ok)
+	assert.Nil(t, condition)
+
+	// call the post step processors
```
Comment (squakez) [inline pkg/trait/knative_test.go:490]: Probably we don't need this in the unit test.

File: pkg/trait/knative_test.go [hunk 6]
```diff
@@ -348,6 +348,155 @@ func TestKnativePlatformHttpDependencies(t *testing.T) {
...
+	knTrait, _ := newKnativeTrait().(*knativeTrait)
+	ok, condition, err = knTrait.Configure(&environment)
+	require.NoError(t, err)
+	assert.True(t, ok)
+	assert.Nil(t, condition)
+
+	// apply the knative trait
+	require.NoError(t, knTrait.Apply(&environment))
+
+	// call the post step processors
```
Comment (squakez) [inline pkg/trait/knative_test.go:417]: Probably we don't need this in the unit test.

File: pkg/trait/knative_test.go (hunk 3 above)
Comment (claudio4j) [reply pkg/trait/knative_test.go]: done

File: pkg/trait/knative_test.go (hunk 5 above)
Comment (squakez) [reply pkg/trait/knative_test.go:489]: Sure, but I don't think it's relevant for the scope of the unit test.

Comment (claudio4j) [reply pkg/trait/knative_test.go:489]: The post processor are required when verifying the dependency, which I just removed earlier. So, makes sense to remove the post processor run.

**Review by squakez — APPROVED** [review]

**Pull request merged**